services/nyx/symbol-filter/src/symbol_filter/filter.py

```diff
@@ -0,0 +1,258 @@
+    LOG.info("Requesting report configuration...")
+    reporter = ReportConfiguration()
+    configuration = reporter.get_report_configuration(filter_id)
+
+    patterns = []
+    for directive in configuration["directives"].splitlines():
+        directive = directive.strip()
+        if not directive or directive.startswith("#"):
+            continue
+
+        if directive.startswith("+:"):
+            pattern = directive[2:].strip()
+            patterns.append((FilterType.INCLUDE, pattern))
+        elif directive.startswith("-:"):
+            pattern = directive[2:].strip()
+            patterns.append((FilterType.EXCLUDE, pattern))
+        else:
+            raise SymbolFilterException("Invalid filter type directive!")
+
+    return patterns
+
+
+def load_path_map() -> dict[str, str]:
+    """Load the path mapping from mozsearch-distinclude.map format."""
+    mapping = {}
+
+    LOG.info("Downloading include map...")
+    flags = BuildFlags(searchfox=True)
+    fetcher = Fetcher("central", "latest", flags, ["searchfox"])
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        url = fetcher.artifact_url(MOZSEARCH_ARTIFACT)
+        temp_file = Path(tmp_dir) / MOZSEARCH_ARTIFACT
+        download_url(url, outfile=temp_file)
+
+        with open(temp_file, encoding="utf-8") as f:
+            lines = f.readlines()
+
+        for line in lines:
+            line = line.strip()
+            if not line:
+                continue
+
+            # Format: <type><dist_path><source_path> (tab-separated with \x1f)
+            if SOURCE_PREFIX in line:
+                parts = line.split("\x1f")
+                assert len(parts) == 3, f"Malformed line in {MOZSEARCH_ARTIFACT}"
+                _, dist_path, source_path = parts
+                mapping[dist_path] = source_path
+
+    return mapping
+
+
+def matches_pattern(path: str, pattern: str) -> bool:
+    """
+    Determine if the supplied path matches the given pattern.
+    :param path: The path to evaluate.
+    :param pattern: The pattern to match.
+    """
+    # Escape special regex characters except * and **
+    # Replace ** with a placeholder first
+    pattern = pattern.replace("**", "\x00")
+    pattern = re.escape(pattern)
+    pattern = pattern.replace("\x00", ".*")  # ** matches anything including /
+    pattern = pattern.replace(r"\*", "[^/]*")  # * matches anything except /
+    return re.match(f"^{pattern}$", path) is not None
+
+
+def should_include_path(path: str, patterns: list[FilterPattern]) -> bool:
+    """
+    Determine if a path should be included based on filter patterns.
+    :param path: The path to evaluate.
+    :param patterns: The patterns to match.
+    """
+    included = False
+    for action, pattern in patterns:
+        if matches_pattern(path, pattern):
+            if action == FilterType.INCLUDE:
+                included = True
+            elif action == FilterType.EXCLUDE:
+                included = False
+
+    return included
```

Comment:
You could shortcut this if you reverse `patterns`:
```suggestion
    # last matching pattern wins
    for action, pattern in reversed(patterns):
        if matches_pattern(path, pattern):
            if action == FilterType.INCLUDE:
                return True
            elif action == FilterType.EXCLUDE:
                return False

    return False
```